cg/services/orders/store_order_services/store_case_order.py

```diff
@@ -34,263 +38,182 @@ def __init__(
-            case_samples: dict[str, Sample] = {}
+                self._update_case_panel(panels=getattr(case, "panels", []), case=status_db_case)
+
             status_db_order.cases.append(status_db_case)
-            for sample in case["samples"]:
-                existing_sample: Sample = self.status_db.get_sample_by_internal_id(
-                    internal_id=sample["internal_id"]
-                )
-                if not existing_sample:
-                    new_sample: Sample = self._create_sample(
-                        case=case,
-                        customer_obj=customer,
-                        order=order,
-                        ordered=ordered,
-                        sample=sample,
-                        ticket=ticket_id,
+            if case.is_new:
+                for sample in case.samples:
+                    sample_mother: DbSample = case_samples.get(
+                        getattr(sample, Pedigree.MOTHER, None)
                     )
-                    case_samples[sample["name"]] = new_sample
-                else:
-                    case_samples[sample["name"]] = existing_sample
-
-            for sample in case["samples"]:
-                sample_mother: Sample = case_samples.get(sample.get(Pedigree.MOTHER))
-                sample_father: Sample = case_samples.get(sample.get(Pedigree.FATHER))
-                with self.status_db.session.no_autoflush:
-                    case_sample: CaseSample = self.status_db.get_case_sample_link(
-                        case_internal_id=status_db_case.internal_id,
-                        sample_internal_id=sample["internal_id"],
+                    sample_father: DbSample = case_samples.get(
+                        getattr(sample, Pedigree.FATHER, None)
                     )
-                if not case_sample:
                     case_sample: CaseSample = self._create_link(
-                        case_obj=status_db_case,
+                        case=status_db_case,
                         family_samples=case_samples,
-                        father_obj=sample_father,
-                        mother_obj=sample_mother,
+                        father=sample_father,
+                        mother=sample_mother,
                         sample=sample,
                     )
 
-                self._update_relationship(
-                    father_obj=sample_father,
-                    link_obj=case_sample,
-                    mother_obj=sample_mother,
-                    sample=sample,
-                )
+                    self._update_relationship(
+                        father=sample_father,
+                        link=case_sample,
+                        mother=sample_mother,
+                        sample=sample,
+                    )
             self.status_db.session.add_all(new_cases)
             self.status_db.session.add(status_db_order)
             self.status_db.session.commit()
```

Comment:
@ChrOertlin suggested decoupling the storing services from the third-party db handling library (SQLAlchemy) by using the functions defined in `cg/store/base.py`. I think it is a good point

Reply:
But do they not commit already? This way we can revert the commit if it crashes down the line